---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

### New Imputation for EMBRAPA with DArT Marker only and DArT plus GBS Markers

Diversity Array Technology LTDA joint all the genotyping data from the four Genotyping Orders that EMBRAPA requested during these six or seven years in one huge file.

So let's check what we got at the DArT report for EMBRAPA DArT genotyping of 2022
```{r, eval = F}
library(genomicMateSelectR)
dir("data/DArT2022")
nskipvcf <- 2
nskipcounts <- 2
VCF2022 <- read.table(here::here("data", "DArT2022", "Report_6902_VCF_Ref_Version6.txt"),
                      sep = "\t", header = T, skip = nskipvcf, comment.char = "",
                      check.names = F)
Counts2022 <- read.table(here::here("data", "DArT2022",
                                    "SEQ_SNPs_counts_0_Target_extend_Ref.csv"),
                         sep = ",", header = T, skip = nskipcounts, check.names = F)
Counts2022[1:10,1:50]
VCF2022[1:10,1:30]
genomicMateSelectR::convertDart2vcf(dartvcfInput = here::here("data", "DArT2022", "Report_6902_VCF_Ref_Version6.txt"),
                                    dartcountsInput = here::here("data", "DArT2022", "SEQ_SNPs_counts_0_Target_extend_Ref.csv"),
                                    nskipvcf = 2, nskipcounts = 2,
                                    outName = "output/DCas22_6902", ncores = 1)
```

Add the VCF header to the DArT VCF file
```{r Add VCF header, eval = F}
system(paste0("bgzip -d -@ 7 output/DArT2022/DCas22_6902_DArTseqLD_AllSites_AllChrom_raw.vcf.gz > ",
              "output/DArT2022/DCas22_6902_DArTseqLD_AllSites_AllChrom_raw.vcf"))

DArTVCFFileSel <- read.table(file = here::here("output", "DArT2022",
                                               "DCas22_6902_DArTseqLD_AllSites_AllChrom_raw.vcf"),
                             sep = "\t", header = T, comment.char = "", check.names = F)

header <- c("##fileformat=VCFv4.0",
            "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">",
            "##FORMAT=<ID=AD,Number=.,Type=Integer,Description=\"Allelic depths for the reference and alternate alleles in the order listed\">",
            "##FORMAT=<ID=DP,Number=1,Type=Integer,Description=\"Read Depth (only filtered reads used for calling)\">",
            "##FORMAT=<ID=PL,Number=3,Type=Float,Description=\"Normalized, Phred-scaled likelihoods for AA,AB,BB genotypes where A=ref and B=alt; not applicable if site is not biallelic\">")

write_lines(header, file = here::here("output", "DCas22_6902",
                                      "DCas22_6902_DArTseqLD_AllSites_AllChrom_raw.vcf"),
            append = F)
write.table(x = DArTVCFFileSel, file = here::here("output", "DCas22_6902",
                                                  "DCas22_6902_DArTseqLD_AllSites_AllChrom_raw.vcf"),
            quote = F, row.names = F, append = T, col.names = T, sep = "\t")

system(paste0("bgzip -c -@ 7 output/DCas22_6902/DCas22_6902_DArTseqLD_AllSites_AllChrom_raw.vcf > ",
              "output/DCas22_6902/DCas22_6902_DArTseqLD_AllSites_AllChrom_raw.vcf.gz"))
```


### Filtering the DArT markers previously the Beagle imputation

```{r, eval = F}
library(here); library(tidyverse)
library(magrittr); library(dplyr)

## Parameters for the Filter function
inPath <- "output/"
inName <- "DCas22_6902_DArTseqLD_AllSites_AllChrom_raw"
outPath <- "output/"
outName <- "DCas22_6902_DArTseqLD_AllSites_AllChrom_rawFiltered"

FilterLuc <- function(inPath = NULL, inName, outPath = NULL, outName, CRthresh = 0.6){
    system(paste0("vcftools --gzvcf ", inPath, inName, ".vcf.gz --freq2 --out ", 
        outPath, inName))
    system(paste0("vcftools --gzvcf ", inPath, inName, ".vcf.gz --missing-site --out ", 
        outPath, inName))
    
    INFO <- read.table(paste0(outPath, inName, ".frq"), stringsAsFactors = F, 
        header = F, skip = 1) %>%
      rename(CHROM = V1, POS = V2, N_ALLELES = V3,
             N_CHR = V4, FREQ1 = V5, FREQ2 = V6)
    callrate <- read.table(paste0(outPath, inName, ".lmiss"), stringsAsFactors = F, 
        header = T) %>% dplyr::select(CHR, POS, N_DATA, F_MISS) %>%
      mutate(CHROM = CHR,
             CR = 1 - F_MISS,
             .keep = "unused")
    
    stats2filterOn <- left_join(INFO, callrate)
    stats2filterOn %<>% dplyr::mutate(FREQ2 = as.numeric(FREQ2)) %>% 
        dplyr::mutate(MAF = ifelse(FREQ2 > 0.5,
                                   yes = 1 - FREQ2, no = FREQ2)) %>% 
      dplyr::select(-FREQ1, -FREQ2)
    
    MAFthresh <- (1/max(stats2filterOn$N_DATA, na.rm = T))**2
    
    sitesPassingFilters <- stats2filterOn %>%
      dplyr::filter(MAF >= MAFthresh, CR >= CRthresh) %>%
      dplyr::select(CHROM, POS)
    print(paste0(nrow(sitesPassingFilters), " sites passing filter"))
    write.table(sitesPassingFilters, file = paste0(outPath, inName, 
        ".sitesPassing"), row.names = F, col.names = F, quote = F)
    system(paste0("vcftools --gzvcf ", inPath, inName, ".vcf.gz", 
        " ", "--positions ", outPath, inName, ".sitesPassing", 
        " ", "--recode --stdout | bgzip -c -@ 24 > ", outPath, 
        outName, ".vcf.gz"))
    print(paste0("Filtering Complete: ", outName))
}

FilterLuc(inPath=inPath, inName=inName,
          outPath=outPath, outName=outName,
          CRthresh = 0.6)
```

##### Download of the Raw Filtered VCF

```{bash, eval = F}
cd output/DArT2022

scp lbraatz@cbsulm35.biohpc.cornell.edu:/workdir/lbraatz/DCas22_6902/output/DCas22_6902_DArTseqLD_AllSites_AllChrom_rawFiltered.vcf.gz .

scp lbraatz@cbsulm35.biohpc.cornell.edu:/workdir/lbraatz/DCas22_6902/output/DCas22_6902_DArTseqLD_AllSites_AllChrom_raw.lmiss .

scp lbraatz@cbsulm35.biohpc.cornell.edu:/workdir/lbraatz/DCas22_6902/output/DCas22_6902_DArTseqLD_AllSites_AllChrom_raw.frq .

cd ../..
```

##### MAF and Call Rate of the Raw data

```{r}
library(tidyverse); library(here)
library(reactable)
INFO <- read.table(here::here("output", "DArT2022", "DCas22_6902_DArTseqLD_AllSites_AllChrom_raw.frq"),
                   stringsAsFactors = F, 
                   header = F, skip = 1) %>%
  rename(CHROM = V1, POS = V2, N_ALLELES = V3,
         N_CHR = V4, FREQ1 = V5, FREQ2 = V6)
callrate <- read.table(here::here("output", "DArT2022", "DCas22_6902_DArTseqLD_AllSites_AllChrom_raw.lmiss"),
                       stringsAsFactors = F, 
                       header = T) %>% dplyr::select(CHR, POS, N_DATA, F_MISS) %>%
  mutate(CHROM = CHR,
         CR = 1 - F_MISS,
         .keep = "unused")

stats2filterOn <- left_join(INFO, callrate)

stats2filterOn %<>% dplyr::mutate(FREQ2 = as.numeric(FREQ2)) %>% 
  dplyr::mutate(MAF = ifelse(FREQ2 > 0.5,
                             yes = 1 - FREQ2, no = FREQ2)) %>% 
  dplyr::select(-FREQ1, -FREQ2)
MAFthresh <- (1/max(stats2filterOn$N_DATA, na.rm = T))**2

stats2filterOn %<>% filter(!is.na(CHROM), CR >= 0.6, MAF >= MAFthresh) %>% 
  select(CR, MAF) %>% rename(CallRate = CR) %>% reshape2::melt(.)

stats2filterOn %>% ggplot(aes(x= value)) + 
  geom_density() + facet_grid(~variable, scales = "free_x") + theme_minimal()
```


### Split the VCF per chromosome

```{r, eval = F}
require(furrr); plan(multisession, workers = 18)
options(future.globals.maxSize=+Inf); options(future.rng.onMisuse="ignore")
  
vcfIn<-here::here("output/","DCas22_6902_DArTseqLD_AllSites_AllChrom_rawFiltered.vcf.gz")
filters<-"--minDP 4 --maxDP 50" # because using GT not PL for impute (Beagle5)
outPath<-here::here("output/")
outSuffix<-"DCas22_6902_DArTseqLD_AllSites_AllChrom_rawFiltered"

future_map(1:18,
           ~genomicMateSelectR::splitVCFbyChr(Chr=.,
                                              vcfIn=vcfIn,filters=filters,
                                              outPath=outPath,
                                              outSuffix=outSuffix))
plan(sequential)

```

### Check for Duplicates in DArT report and GBS VCF file - [link](Duplicates.html)

### Imputation DArT markers only - [link](DArTImp.html)

#### Quality control

| Pre |             | Post |             |
| :-: | :---------: | :--: | :---------: |
| MAF | $≥(2/7827)$ | MAF  | $≥(2/7827)$ |
| CR  | $≥0.6$      | HWE  | $P_{HWE}≥10^{-20}$ |
|     |             |      |             |

#### SNPs Markers

5,914 SNPs Markers from 7,827 clones

| Chr   | N˚Mkrs | Chr    | N˚Mkrs | Chr    | N˚Mkrs |
| :---: | :----: | :----: | :----: | :----: | :----: |
| Chr 1 | 674    | Chr 7  | 216    | Chr 13 | 261    |
| Chr 2 | 369    | Chr 8  | 294    | Chr 14 | 380    |
| Chr 3 | 372    | Chr 9  | 262    | Chr 15 | 297    |
| Chr 4 | 371    | Chr 10 | 301    | Chr 16 | 219    |
| Chr 5 | 337    | Chr 11 | 317    | Chr 17 | 305    |
| Chr 6 | 379    | Chr 12 | 285    | Chr 18 | 275    |
|       |        |        |        |        |        |

### Imputation GBS + DArT Markers - [link](GBS_DArTImp.html)

#### Quality control

| Pre |             | Post |             |
| :-: | :---------: | :--: | :---------: |
| MAF | $≥(2/7827)$ | MAF  | $≥(2/7827)$ |
| CR  | $≥0.6$      | HWE  | $P_{HWE}≥10^{-20}$ |
| HWE | $P_{HWE}≥10^{-20}$ |      |             |
|     |             |      |      |

#### SNPs Markers

37,877 SNPs Markers from 7,827 clones

| Chr   | N˚Mkrs | Chr    | N˚Mkrs | Chr    | N˚Mkrs |
| :---: | :----: | :----: | :----: | :----: | :----: |
| Chr 1 | 3,669  | Chr 7  | 1,256  | Chr 13 | 1,812  |
| Chr 2 | 2,717  | Chr 8  | 2,015  | Chr 14 | 2,105  |
| Chr 3 | 2,547  | Chr 9  | 2,117  | Chr 15 | 2,428  |
| Chr 4 | 1,982  | Chr 10 | 1,519  | Chr 16 | 1,692  |
| Chr 5 | 2,607  | Chr 11 | 2,109  | Chr 17 | 1,542  |
| Chr 6 | 2,497  | Chr 12 | 1,733  | Chr 18 | 1,530  |
|       |        |        |        |        |        |

## Results of the Imputations - [link](CheckImp.html)


## Genomic Prediction Analysis for Yield traits

### Genotypic data preparation - [link](PrepareGenData.html)
