---
title: "Preparing Genotype Data"
author: "LucianoRogerio"
date: "2022-06-01"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Preparing the Genotype data for the Genomic prediction

```{r Preparing the Training population gen data}
library(tidyverse); library(data.table); library(sommer)
TPClonesPhen <- read.table(file = here::here("output", "BRTP_Phenotyping2022.txt"),
                           sep = "\t", header = F) %>% rename(Clone = V1)
DArTClonesGen <- read.table(file = here::here("data", "DArTGPInfo2.csv"),
                         sep = ",", header = T)
TPDArTRefPopSel <- read.table(file = here::here("output", "DCas22_6902", "DArTClonesVerifiedIDB.txt"),
                              sep = "\t", header = F)

DArTClonesGenUn <- DArTClonesGen[!duplicated(DArTClonesGen$Observation_unit_name),]
DArTClonesGenDup <- DArTClonesGen[duplicated(DArTClonesGen$Observation_unit_name),]
DArTClonesGenDupSel <- DArTClonesGenDup[DArTClonesGenDup$DarTMatch %in% TPDArTRefPopSel$V1,]
DArTClonesGenDupSel2 <- DArTClonesGenDup[!(DArTClonesGenDup$Observation_unit_name %in%
                                             DArTClonesGenDupSel$Observation_unit_name),] %>%
  .[!duplicated(.[,"Observation_unit_name"]),]

DArTClonesGen2 <- rbind(DArTClonesGenUn, DArTClonesGenDupSel, DArTClonesGenDupSel2)
TrainingPop <- DArTClonesGen2[DArTClonesGen2$Observation_unit_name %in% TPClonesPhen$Clone,]
StudyPop <- DArTClonesGen2[DArTClonesGen2$Observation_unit_name %like% "BR-21",]

GSPop <- rbind(TrainingPop, StudyPop)

DArTsnps <- readRDS(file = here::here("output", "DArT2022", "DCas22_DArT_ReadyForGP_Dos.rds"))
GBSDArTsnps <- readRDS(file = here::here("output", "DCas22_6902", "DCas22_GBSandDArT_ReadyForGP_Dos.rds"))

GSPop <- GSPop[GSPop$DarTMatch %in% rownames(DArTsnps),] %>%
  .[order(match(.[,"DarTMatch"], rownames(DArTsnps))),] %>%
  filter(!Observation_unit_name %like% "_conf") %>%
  group_by(Observation_unit_name) %>% slice(1)

DArTGSPop <- DArTsnps[rownames(DArTsnps) %in% GSPop$DarTMatch,]
GBSDArTGSPop <- GBSDArTsnps[rownames(GBSDArTsnps) %in% GSPop$DarTMatch,]

dim(DArTGSPop); dim(GBSDArTGSPop)

GBSDArTGSPop <- GBSDArTGSPop[order(match(rownames(GBSDArTGSPop),rownames(DArTGSPop))),]
GSPop <- GSPop[order(match(GSPop$DarTMatch, rownames(DArTGSPop))),]

rownames(DArTGSPop) <- GSPop$Observation_unit_name
rownames(GBSDArTGSPop) <- GSPop$Observation_unit_name

MAFD <- tibble(Freq = colMeans(DArTGSPop)) %>%
  mutate(MAF = ifelse(Freq > 0.5, yes = 1 - Freq, no = Freq),
         .keep = "unused")
MAFGD <- tibble(Freq = colMeans(GBSDArTGSPop)) %>%
  mutate(MAF = ifelse(Freq > 0.5, yes = 1 - Freq, no = Freq),
         .keep = "unused")

DArTGSPopSel <- DArTGSPop[ , MAFD >= 0.01]
GBSDArTGSPopSel <- GBSDArTGSPop[ , MAFGD >= 0.01]
dim(DArTGSPopSel); dim(GBSDArTGSPopSel)

saveRDS(file = here::here("output", "DCas22_6902", "DCas22_GBSandDArT_ReadyForGP_DosFilt1per.rds"),
        object = GBSDArTGSPopSel)
GBSMkrsSel <- colnames(GBSDArTGSPopSel) %>% strsplit(split = "_") %>% unlist %>%
  matrix(ncol = 3, byrow = T)
colnames(GBSMkrsSel) <- c("Chr", "Pos", "Ref")
GBSMkrsSel <- GBSMkrsSel %>% as_tibble %>% mutate(Pos = as.integer(Pos),
                                                  Chr = gsub(pattern = "S", replacement = "", x = .$Chr) %>% 
                                                    as.integer)
head(GBSMkrsSel)

write.table(x = GBSMkrsSel, file = here::here("output", "DCas22_6902", "GBSMkrsSelGP"),
            quote = FALSE, row.names = F, col.names = F)
write.table(x = GSPop$DarTMatch, file = here::here("output", "DCas22_6902", "GBSCloneSelGP"),
            quote = FALSE, row.names = F, col.names = F)


GMD <- kinship(DArTGSPopSel, type = "add")
GMDD <- kinship(DArTGSPopSel, type = "domClassic")
GMGD <- kinship(GBSDArTGSPopSel, type = "add")
GMGDD <- kinship(GBSDArTGSPopSel, type = "domClassic")

saveRDS(GMD, file = here::here("output", "GAddmatrixDArTGS.rds"))
saveRDS(GMDD, file = here::here("output", "GDommatrixDArTGS.rds"))
saveRDS(GMGD, file = here::here("output", "GAddmatrixGBSandDArTGS.rds"))
saveRDS(GMGD, file = here::here("output", "GDommatrixGBSandDArTGS.rds"))
```


### Fig 1. DArT Density Markers after the filter of MAF 0.05 for the GS Population
```{r, echo = F}
library(CMplot)
CMsnps <- tibble(SNP = colnames(DArTGSPopSel),
                 chr = substring(SNP,1,3),
                 pos = substring(SNP,4)) %>%
  mutate(chr = gsub(pattern = "_", replacement = "", x = chr) %>%
           gsub(pattern = "S", replacement = "") %>% as.integer,
         pos = gsub(pattern = "_[A-Z]", replacement = "", x = pos) %>%
           gsub(pattern = "_", replacement = "", x = .) %>% as.integer)
CMplot(CMsnps, plot.type = "d", bin.size = 1e6, col = c("darkgreen", "yellow", "red"),
       file = "jpg", memo = "DArTGSDensityMkrs", dpi = 500, file.output = F, verbose = TRUE)
```

### Fig 2. GBS+DArT Density Markers after the filter of MAF 0.05 for the GS Population
```{r, echo = F}
library(CMplot)
CMsnps <- tibble(SNP = colnames(GBSDArTGSPopSel),
                 chr = substring(SNP,1,3),
                 pos = substring(SNP,4)) %>%
  mutate(chr = gsub(pattern = "_", replacement = "", x = chr) %>%
           gsub(pattern = "S", replacement = "") %>% as.integer,
         pos = gsub(pattern = "_[A-Z]", replacement = "", x = pos) %>%
           gsub(pattern = "_", replacement = "", x = .) %>% as.integer)
CMplot(CMsnps, plot.type = "d", bin.size = 1e6, col = c("darkgreen", "yellow", "red"),
       file = "jpg", memo = "GBSDArTGSDensityMkrs", dpi = 500, file.output = F, verbose = TRUE)
```


### Fig 3. Correlogram of diagonal index of the Genotyping platforms
```{r Correlogram Diag, echo = F}
library(GGally)
RefPopImp <- DArTClonesGen[DArTClonesGen$DarTMatch %in% TPDArTRefPopSel$V1,]
DiagPlot <- tibble(clone = row.names(GMD),
                   Pop = factor(ifelse(clone %like% "BR-21",
                                yes = "C3",
                                no = ifelse(clone %like% "BR-18",
                                            yes = "C1",
                                            no = ifelse(clone %like% "BR-19",
                                                        yes = "C2",
                                                        no = ifelse(clone %in% RefPopImp$Observation_unit_name,
                                                                    yes = "RefPop",
                                                                    no = "C0")))),
                                levels = c("RefPop", "C0", "C1", "C2", "C3")),
                   DArT = diag(GMD),
                   GBSDArT = diag(GMGD))

ggpairs(DiagPlot %>% dplyr::select(DArT, GBSDArT))
```

### Fig 4. Scatter plot of the diagonal index from G matrix
```{r Scatter plot of diagonal index from G matrix, echo = F}
DiagPlot %>% ggplot(aes(x = DArT, y = GBSDArT, color = Pop)) + 
  geom_point(size = 0.5, alpha = 0.6) + theme_bw() + ylim(0.5,2) + xlim(0.5,2) +
  guides(color = guide_legend(override.aes = list(size = 2)))
```

### Fig 5. Correlogram of the out diagonal index of the Genotyping platforms
```{r Correlogram OutDiag, echo = F}
OutDiagPlot <- tibble(DArT = GMD[upper.tri(GMD,diag = F)],
                      GBSDArT = GMGD[upper.tri(GMGD, diag = F)])

ggpairs(OutDiagPlot)
```

## Prepare the Haplotype file for the genomic Mate Selection

### Create a Haplotype file
```{r, eval = F}
### Filtering the VCF file for the markers used on the Genomic Prediction
system(paste0("vcftools --gzvcf output/DCas22_6902/DCas22_6902_RefandStudyPopimputedAndFiltered.vcf.gz ",
              "--keep output/DCas22_6902/GBSCloneSelGP ",
              "--positions output/DCas22_6902/GBSMkrsSelGP ",
              "--recode --stdout | ",
              "bgzip -c -@ 7 > ",
              "output/DCas22_6902/DCas22_6902_RefandStudyPopimputedAndFiltered4GP.vcf.gz"))

### create the Haplotype matrix from the VCF file

system(paste0("tabix -f -p vcf output/DCas22_6902/",
              "DCas22_6902_RefandStudyPopimputedAndFiltered4GP.vcf.gz"))

system(paste0("bcftools convert --hapsample output/HapDCas22_69024GMS ",
              "output/DCas22_6902/DCas22_6902_RefandStudyPopimputedAndFiltered4GP.vcf.gz"))
```

```{r, eval = F}
library(genomicMateSelectR)
library(data.table)

vcfName <- "HapDCas22_69024GMS"
haps <- fread(paste0("output/",vcfName,".hap.gz"),
              stringsAsFactors = F,header = F) %>% 
  as.data.frame
sampleids <- fread(paste0("output/",vcfName,".samples"),
                   stringsAsFactors = F,header = F,skip = 2) %>% 
  as.data.frame

SampleIdsDArT <- GSPop[order(match(GSPop$DarTMatch, sampleids$V1)),]

sampleids2 <- tibble(V1 = SampleIdsDArT$Observation_unit_name,
                      V2 = SampleIdsDArT$Observation_unit_name)
hapids <- sampleids2 %>% 
  dplyr::select(V1,V2) %>% 
  mutate(SampleIndex=1:nrow(.)) %>% 
  rename(HapA=V1,HapB=V2) %>% 
  pivot_longer(cols=c(HapA,HapB),
               names_to = "Haplo",values_to = "SampleID") %>% 
  mutate(HapID=paste0(SampleID,"_",Haplo)) %>% 
  arrange(SampleIndex)
colnames(haps)<-c("Chr","HAP_ID","Pos","REF","ALT",hapids$HapID)
dim(haps)

haps %<>% 
     mutate(HAP_ID=gsub(pattern = ":", replacement = "_", x = HAP_ID)) %>% 
     column_to_rownames(var = "HAP_ID") %>% 
     dplyr::select(-Chr,-Pos,-REF,-ALT) %>% 
     t %>% 
     as.matrix(.)
dim(haps)
```

### Prepare a dosage file with the Hap file

```{r}
dosages <- haps %>%
     as.data.frame(.) %>% 
     rownames_to_column(var = "GID") %>% 
     separate(GID,c("SampleID","Haplo"),"_Hap",remove = T) %>% 
     dplyr::select(-Haplo) %>% 
     group_by(SampleID) %>% 
     summarise(across(everything(),~sum(.))) %>% 
     ungroup() %>% 
     column_to_rownames(var = "SampleID") %>% 
     as.matrix %>% 
     # preserve same order as in haps
     .[sampleids2$V1,]
dim(dosages)

dosages[1:5,1:5]

### Filter

dosages<-maf_filter(dosages,thresh = 0.01)
dim(dosages)
#> [1]  963 3986
# subset haps to match
haps<-haps[,colnames(dosages)]

saveRDS(dosages,file=here::here("output","dosagesEmbrapa2022.rds"))
saveRDS(haps,file=here::here("output","haplotypesEmbrapa2022.rds"))
```

## Recombination Frequency matrix

```{r, eval = T}
genmap<-read.table("https://cassavabase.org/ftp/marnin_datasets/NGC_BigData/CassavaGeneticMap/cassava_cM_pred.v6.allchr.txt",
                   header = F, sep=';', stringsAsFactors = F) %>% 
     rename(SNP_ID=V1,Pos=V2,cM=V3) %>% 
  as_tibble
genmap %>% dim
#> [1] 120979      3

snps_genmap<-tibble(DoseSNP_ID=colnames(dosages)) %>% 
     separate(DoseSNP_ID,c("Chr","Pos","Ref","Alt"),remove = F) %>% 
     mutate(SNP_ID=paste0("S",Chr,"_",Pos)) %>% 
     full_join(genmap %>% 
                    separate(SNP_ID,c("Chr","POS"),"_",remove = F) %>% 
                    dplyr::select(-POS) %>% 
                    mutate(Chr=gsub("S","",Chr)) %>% 
                    mutate(across(everything(),as.character)))
#> Joining, by = c("Chr", "Pos", "SNP_ID")

interpolate_genmap<-function(data){
  # for each chromosome map
  # find and _decrements_ in the genetic map distance
  # fix them to the cumulative max to force map to be only increasing
  # fit a spline for each chromosome
  # Use it to predict values for positions not previously on the map
  # fix them AGAIN (in case) to the cumulative max, forcing map to only increase
  data_forspline<-data %>% 
    filter(!is.na(cM)) %>% 
    mutate(cumMax=cummax(cM),
           cumIncrement=cM-cumMax) %>% 
    filter(cumIncrement>=0) %>% 
    dplyr::select(-cumMax,-cumIncrement)
  
  spline<-data_forspline %$% smooth.spline(x=Pos,y=cM,spar = 0.75)
  
  splinemap<-predict(spline,x = data$Pos) %>% 
    as_tibble(.) %>% 
    rename(Pos=x,cM=y) %>% 
    mutate(cumMax=cummax(cM),
           cumIncrement=cM-cumMax) %>% 
    mutate(cM=cumMax) %>% 
    dplyr::select(-cumMax,-cumIncrement)
  
  return(splinemap) 
}

splined_snps_genmap <- snps_genmap %>% 
  dplyr::select(-cM) %>% 
  mutate(Pos=as.numeric(Pos)) %>% 
  left_join(snps_genmap %>% 
              mutate(across(c(Pos,cM),as.numeric)) %>% 
              arrange(Chr,Pos) %>% 
              nest(-Chr) %>% 
              mutate(data=map(data,interpolate_genmap)) %>% 
              unnest(data)) %>% 
  distinct
#> Warning: All elements of `...` must be named.
#> Did you want `data = c(DoseSNP_ID, Pos, Ref, Alt, SNP_ID, cM)`?
#> Joining, by = c("Chr", "Pos")

all(colnames(dosages) %in% splined_snps_genmap$DoseSNP_ID)
#> [1] TRUE

splined_snps_genmap %>% 
     filter(DoseSNP_ID %in% colnames(dosages)) %>% 
     saveRDS(.,file=here::here("output","interpolated_genmap.rds"))

genmap<-readRDS(file=here::here("output","interpolated_genmap.rds"))
m<-genmap$cM;
names(m)<-genmap$DoseSNP_ID
recombFreqMat<-1-(2*genmap2recombfreq(m,nChr = 18))
saveRDS(recombFreqMat,file=here::here("output","recombFreqMat_1minus2c.rds"))
```

